src/shared/parameter_v2/config/examples/parameter.c

```diff
@@ -0,0 +1,111 @@
+#include "shared/parameter_v2/config/examples/parameter.h"
+
+#include <memory.h>
+#include <stdlib.h>
+
+typedef struct BoolParameter
+{
+    const bool value;
+} BoolParameter_t;
+
+typedef struct IntParameter
+{
+    const int value;
+} IntParameter_t;
+
+typedef struct UIntParameter
+{
+    const unsigned value;
+} UIntParameter_t;
+
+typedef struct FloatParameter
+{
+    const float value;
+} FloatParameter_t;
+
+typedef struct StringParameter
+{
+    const char* value;
+} StringParameter_t;
+
+
+float app_dynamic_parameters_getFloat(const FloatParameter_t* param)
+{
+    return param->value;
+}
+int app_dynamic_parameters_getInt(const IntParameter_t* param)
+{
+    return param->value;
+}
+unsigned app_dynamic_parameters_getUInt(const UIntParameter_t* param)
+{
+    return param->value;
+}
+const char* app_dynamic_parameters_getString(const StringParameter_t* param)
+{
+    return param->value;
+}
+bool app_dynamic_parameters_getBool(const BoolParameter_t* param)
+{
+    return param->value;
+}
+
+const FloatParameter_t* app_dynamic_parameters_createFloatParameter(float value)
+{
+    FloatParameter_t* param     = (FloatParameter_t*)malloc(sizeof(FloatParameter_t));
+    FloatParameter_t param_init = {.value = value};
+    memcpy(param, &param_init, sizeof(FloatParameter_t));
```

Comment:
this is the safest way to do set initialize "const" ptrs after malloc without hitting undefined behaviour

https://stackoverflow.com/questions/9691404/how-to-initialize-const-in-a-struct-in-c-with-malloc

Reply:
Can we add this as a comment?

Reply:
added